src/feature/catalog/api/fetch-products.ts

```diff
@@ -1,8 +1,15 @@
-import type { Product } from '@commercetools/platform-sdk';
+import type { ProductProjection } from '@commercetools/platform-sdk';
 
 import apiRoot from '@/feature/api/api-client-credentials-flow';
 
-export const fetchProducts = async (): Promise<Product[]> => {
-  const response = await apiRoot.products().get().execute();
+export const fetchProducts = async (categoryId?: string): Promise<ProductProjection[]> => {
+  const response = await apiRoot
+    .productProjections()
+    .get({
+      queryArgs: {
+        where: categoryId ? `categories(id="${categoryId}")` : undefined,
+      },
+    })
+    .execute();
   return response.body.results;
 };
```

Comment:
```suggestion
export const fetchProducts = async (categoryId?: string): Promise<Product[]> => {
  const response = await apiRoot
    .products()
    .get({
      queryArgs: {
        where: categoryId ? `masterData(current(categories(id="${categoryId}")))` : undefined,
      },
    })
    .execute();
  return response.body.results;
};
```

Можно оставить `Products`, просто категории лежат глубже в дереве

Reply:
А нам надо Products? как будто можно и то и то подходит просто ProductProjection удобнее и не надо masterData

Reply:
да как будто без разницы, да.

В целом я согласен, в нашем текущем кейсе даже проще обращаться к `productProjection`, потому что судя по [доке ](https://docs.commercetools.com/api/projects/productProjections)это конечный (опубликованный) вариант товара без `current/staged`, так что можно оставить и так, равно как и я у себя не буду менять)